core/domain/expression_parser.py

```diff
@@ -0,0 +1,509 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A parser that generates a parse tree for math expressions.
+
+It uses the following grammar in Backus-Naur form:
+
+<expr> ::= <expr2> (('+' | '-') <expr2>)*
+<expr2> ::= <expr1> (('*' | '/') <expr1>)*
```

Comment:
Can you please give these names which have more semantic meaning than "expr1" and "expr2"?

Reply:
Done.